--- lib/src/whip_client/whip_client_web.dart
+++ lib/src/whip_client/whip_client_web.dart
@@ -231,8 +231,17 @@
 }

+@JS("window")
+external JSObject get _window;
+
+@JS()
+@anonymous
+extension type _WindowWHIPExt(JSObject _) implements JSObject {
+  external JSAny? get WHIPClient;
+}
+
 /// Library status check.
 bool get isWHIPLibraryLoaded =>
-    web.window.hasProperty("WHIPClient".toJS).toDart;
+    !(_window as _WindowWHIPExt).WHIPClient.isUndefinedOrNull;

 // Cache the initialization future to prevent redundant <script> injections
